src/test/java/net/oauth/jsontoken/AsyncJsonTokenParserTest.java

```diff
@@ -129,17 +127,14 @@ private AsyncJsonTokenParser getAsyncJsonTokenParser(AsyncVerifierProviders prov
     return new AsyncJsonTokenParser(clock, providers, executor, checkers);
   }
 
-  private <T extends Throwable> void assertCause(Class<T> throwableClass, Callable func) throws Exception {
+  private <T extends Throwable> void assertFailsWithCause(Class<T> throwableClass, ThrowingRunnable runnable) {
     try {
```

Comment:
There was one more part of my previous comment: You can simplify this by using assertThrows(ExecutionException) instead of the try-catch. It will also return the exception which you can use to check the cause.

Reply:
Sorry, I missed it. Should be fixed now.